src/org/labkey/test/tests/list/ListIndexingTest.java

```diff
@@ -0,0 +1,186 @@
+
+import org.jetbrains.annotations.Nullable;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.TestFileUtils;
+import org.labkey.test.categories.Daily;
+import org.labkey.test.components.list.AdvancedListSettingsDialog;
+import org.labkey.test.pages.list.EditListDefinitionPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.util.ListHelper;
+import org.labkey.test.util.PortalHelper;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@Category({Daily.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 5)
+public class ListIndexingTest extends BaseWebDriverTest
+{
+    private final ListHelper _listHelper = new ListHelper(this);
+    private static final String listName = "NIMHDemographics";
+
+    @Override
+    protected @Nullable String getProjectName()
+    {
+        return getClass().getSimpleName() + " Project";
+    }
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return Arrays.asList("list");
+    }
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        ListIndexingTest init = (ListIndexingTest) getCurrentTest();
+        init.doSetup();
+    }
+
+    /*
+       Regression test cases for Issue 48056: More list indexing fixes
+     */
+    private void doSetup()
+    {
+        log("Setup project and list module");
+        _containerHelper.createProject(getProjectName(), null);
+        _listHelper.importListArchive(getProjectName(), TestFileUtils.getSampleData("lists/ListDemo.lists.zip"));
+
+        goToProjectHome();
+        new PortalHelper(this).addWebPart("Search");
+    }
+
+    @Test
+    public void testEachItemIndexing()
+    {
+        goToProjectHome();
+        EditListDefinitionPage editListDefinitionPage = _listHelper.goToEditDesign("NIMHSlides");
+        editListDefinitionPage.openAdvancedListSettings()
+                .indexEachItemAsASeparateDocument(true, "Subject Id: ${SubjectId}", AdvancedListSettingsDialog.SearchIndexOptions.NonPhiFields)
+                .clickApply()
+                .clickSave();
+
+        searchFor(getProjectName(), "10001", 8, null);
+    }
+
+    @Test
+    public void testEntireListIndexing()
+    {
+        /*
+            Test coverage for :
+            Issue 48188: Selecting 'All non-PHI' or 'Custom template' under 'Index each item as a separate document' doesn't stick
+         */
+        goToProjectHome();
+        EditListDefinitionPage editDesign = _listHelper.goToEditDesign(listName);
+        editDesign.openAdvancedListSettings()
+                .indexEntireListAsASingleDocument(true, "",
+                        AdvancedListSettingsDialog.SearchIncludeOptions.MetadataAndData,
+                        AdvancedListSettingsDialog.SearchIndexOptions.CustomTemplate, "${Name}")
```

Comment:
This is confusing... per the PR, help popup, and docs, the entire-list title template doesn't support any substitutions. I believe all the search results will show up with literally `${Name}` as the title. Did you see something different?